### PR TITLE
[TASK] Update build pipeline to use v4 of actions

### PR DESCRIPTION
**Pull request by thomashohn**

### CODE REVIEW

**Review by 43081j — APPROVED** [review]
looks good to me!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -32,10 +32,10 @@ jobs:
     needs: build
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
-          node-version: 22
+          node-version: 20
```
Comment (43081j) [inline .github/workflows/ci.yml:38]: was there a particular reason this one moved down?

Comment (thomashohn) [reply .github/workflows/ci.yml:38]: As i remember the pipelien of x.x.x.x failed when raising til to node 22

Comment (43081j) [reply .github/workflows/ci.yml:38]: i think its just a red herring

this `node-version` is of the `publish-npm` job which didn't run, the v20 in the failure you linked comes from the matrix further up

can you put this line back and we will see if it persistently fails. ill have a dig into it if it does

Comment (thomashohn) [reply .github/workflows/ci.yml]: You seem to have been right - re-added it

**Pull request merged**